exercises/change/example.js

```diff
@@ -0,0 +1,139 @@
+class Change {}
+
+export default Change;
+
+// data structure to hold each candidate solution that is generated
+class Candidate {
+  constructor() {
+    this.wasSearched = false;
+    this.coins = [];
+  }
+
+  searched() {
+    this.wasSearched = true;
+  }
+
+  isSearched() {
+    return this.wasSearched;
+  }
+
+  getCoins() {
+    return this.coins;
+  }
+
+  addCoin(coin) {
+    const sortNum = (a, b) => a - b;
+
+    this.coins.push(coin);
+    this.coins.sort(sortNum);
+  }
+
+  getCoinCount() {
+    return this.coins.length;
+  }
+
+  getSum() {
+    const getSum = (total, num) => total + num;
+    return this.coins.reduce(getSum);
+  }
+}
+
+Change.prototype.calculate = (coinArray, target) => {
+  const candidates = [];
+  // fill the array with 0 to start
+  candidates[target] = 0;
+  candidates.fill(0);
+
+  // save a new candidate to the candidate array
+  const saveCandidate = (candidate) => {
+    const sum = candidate.getSum();
+
+    if (candidate.getSum() <= target &&
+            typeof (candidates[sum]) !== 'number' &&
+            candidates[sum].getCoinCount() > candidate.getCoinCount()) {
+      candidates[sum] = candidate;
+    }
+
+    if (candidate.getSum() <= target &&
+           typeof (candidates[sum]) === 'number') {
+      candidates[sum] = candidate;
+    }
+  };
+
+  // initialize the candidate array with the given coins only
+  const initialize = () => {
+    for (let j = 0; j < coinArray.length; j += 1) {
+      const temp = coinArray[j];
+      const candidate = new Candidate();
+      candidate.addCoin(temp);
+      saveCandidate(candidate);
+    }
+  };
+
+  // is everthing searched?
+  const isDone = () => {
+    let done = true;
+    for (let i = 0; i < candidates.length; i += 1) {
+      const temp = candidates[i];
+      if (typeof (temp) !== 'number') {
+        if (temp.isSearched() === false) {
+          done = false;
+          break;
+        }
+      }
+    }
+    return done;
+  };
+
+  // get the next unsearched member of the candidate array
+  const getNext = () => {
+    for (let i = 0; i < candidates.length; i += 1) {
+      const temp = candidates[i];
+      if (typeof (temp) !== 'number' &&
+                temp.isSearched() === false) return temp;
+    }
+    return null;
+  };
+
+  // for the candidate, generate another candate for each of the possible coins
+  const branch = (current) => {
+    for (let j = 0; j < coinArray.length; j += 1) {
+      // make a new Candidate for coin type
+      const candidate = new Candidate();
+
+      // copy the curent coins into it and add the new coin type
+      for (let i = 0; i < current.getCoins().length; i += 1) {
+        candidate.addCoin(current.getCoins()[i]);
```

Comment:
Again, you have two loops that use index variables that you don't really need.  Hit it with some ES6!!!